### PR TITLE
[nnyeah] Refactor module usage

### PR DESCRIPTION
**Pull request by stephen-hawley**
Clean up usage of modules in the app.
the var 'module' gets used all over the place. I'm going to refactor to fix up usage of imports and this will make things much easier to track.

In addition, there are a number of types that get used from `module.TypeSystem.SomeType` which I put into locals just to tidy up.

### CODE REVIEW

**Review by chamons — APPROVED** [review]

File: tools/nnyeah/nnyeah/Program.cs [hunk 1]
```diff
@@ -58,15 +58,19 @@ static void Main (string [] args)
...
 		{
 			if (!TryLoadTypeAndModuleMap (xamarinAssembly!, microsoftAssembly!, publicOnly: true,
-				out var typeAndModuleMap, out var failureReason)) {
+				out var typeAndModuleMap, out var failureReason, out var xamarinModule,
+				out var microsoftModule)) {
 				Console.Error.WriteLine (Errors.E0011, failureReason);
 			}
-			ReworkFile (infile!, outfile!, verbose, forceOverwrite, suppressWarnings, typeAndModuleMap!);
+			ReworkFile (infile!, outfile!, verbose, forceOverwrite, suppressWarnings, typeAndModuleMap!,
+				xamarinModule!, microsoftModule!);
```
Comment (mandel-macaque) [inline tools/nnyeah/nnyeah/Program.cs:66]: Inverting the if will remove all those !
```csharp
if (TryLoadTypeAndModuleMap (xamarinAssembly!, microsoftAssembly!, publicOnly: true,
 				out var typeAndModuleMap, out var failureReason)) {
    ReworkFile (infile!, outfile!, verbose, forceOverwrite, suppressWarnings, typeAndModuleMap,
  				xamarinModule, microsoftModule);
} else {
    Console.Error.WriteLine (Errors.E0011, failureReason);
}
```

Give it a go, but not a show stopper.

File: tools/nnyeah/nnyeah/Reworker.cs [hunk 2]
```diff
@@ -53,59 +53,60 @@ public static bool NeedsReworking (ModuleDefinition module)
...
 				NintTypeReference = EmptyTypeReference;
 			}
-			if (!module.TryGetTypeReference ("System.nuint", out NuintTypeReference)) {
+			if (!ModuleToEdit.TryGetTypeReference ("System.nuint", out NuintTypeReference)) {
 				NuintTypeReference = EmptyTypeReference;
 			}
-			if (!module.TryGetTypeReference ("System.nfloat", out NfloatTypeReference)) {
+			if (!ModuleToEdit.TryGetTypeReference ("System.nfloat", out NfloatTypeReference)) {
 				NfloatTypeReference = EmptyTypeReference;
+				NewNfloatTypeReference = EmptyTypeReference;
```
Comment (mandel-macaque) [inline tools/nnyeah/nnyeah/Reworker.cs:101]: Same, we could init all to be the empty ref and then just set them in the out param.

**Review by mandel-macaque — APPROVED** [review]
@stephen-hawley  some things can be written to be more concise. Initialising the out params to the default will make everything less verbose. Not a show stopper atm.

File: tools/nnyeah/nnyeah/Program.cs [hunk 3]
```diff
@@ -108,21 +112,27 @@ static void Main (string [] args)
...
 				result = map;
 				reason = null;
+				xamarinModule = earlierModule;
+				microsoftModule = laterModule;
 				return true;
 			} catch (Exception e) {
 				result = null;
 				reason = e.Message;
+				xamarinModule = null;
+				microsoftModule = null;
```
[the remark this answers is not in the document]
Comment (stephen-hawley) [reply tools/nnyeah/nnyeah/Program.cs:122]: Understood, but I find that doing that can hide certain classes of errors. Keeping locality of the set ensures that setting in the "good" case and in the "evil" case both happen, whereas initializing to `null` at the beginning of the method would hide the error if someone remove the "good" case.

**Pull request merged**